tests/test_oauth2_message.py

```diff
@@ -595,3 +621,100 @@ def test_missing_required(self):
 
         with pytest.raises(MissingRequiredAttribute):
             err.to_urlencoded()
+
+
+def test_to_jwt_rsa():
```

Comment:
Those tests could maybe be simplified into just one function with https://docs.pytest.org/en/latest/parametrize.html#

The body is nearly the same, just a few different parameters.